### PR TITLE
feat: sqlite database implementation

### LINKED ISSUE #1241: epic: SQLite database implementation
Implement a ModelListUtils class base on this discussion #1154 

## Overview

The `ModelListUtils` class provides a robust and thread-safe interface for managing a list of machine learning models. It handles operations such as adding, updating, deleting, and retrieving model information, with a focus on maintaining unique identifiers for each model.

## Key Features

1. **Thread-safe Operations**: Uses mutex locking to ensure thread safety for all operations.
2. **File-based Storage**: Manages model information in a text file, allowing for persistence between program runs.
3. **Unique Identifier Generation**: Automatically generates unique, shortened aliases for models.
4. **CRUD Operations**: Supports Creating, Reading, Updating, and Deleting model entries.
5. **Error Handling**: Robust error checking and exception throwing for invalid operations.

### ModelEntry Struct
Represents a single model entry with fields:
- `model_id`: Unique identifier for the model
- `author_repo_id`: Author or repository identifier
- `branch_name`: Branch name in the repository
- `path_to_model_yaml`: Path to the model's YAML file
- `model_alias`: Shortened alias for the model
- `status`: Current status of the model (READY or RUNNING)

### ModelListUtils Class Methods

1. **LoadModelList()**
   - Loads model entries from the file, creating the file if it doesn't exist.
   - Returns a vector of ModelEntry objects.

2. **SaveModelList()**
   - Saves the current list of model entries to the file.

3. **GetModelInfo(identifier)**
   - Retrieves model information based on model_id or model_alias.
   - Throws an exception if the model is not found.

4. **AddModelEntry(new_entry, use_short_alias)**
   - Adds a new model entry to the list.
   - Optionally generates a shortened alias for the model.
   - Ensures uniqueness of model_id and model_alias.

5. **UpdateModelEntry(identifier, updated_entry)**
   - Updates an existing model entry.
   - Identified by either model_id or model_alias.

6. **DeleteModelEntry(identifier)**
   - Deletes a model entry if it exists and is in READY state.

7. **PrintModelInfo(entry)**
   - Prints detailed information about a model entry.

8. **GenerateShortenedAlias(model_id, entries)**
   - Generates a unique, shortened alias for a model.
   - Uses a hierarchical approach to create aliases, prioritizing brevity while ensuring uniqueness.

## Alias Generation Logic

The `GenerateShortenedAlias` function creates aliases in the following order:
1. Filename only (e.g., `model_id_xxx`)
2. Parent directory + filename (e.g., `llama3.1-7b-gguf:model_id_xxx`)
3. Grandparent directory + parent directory + filename (e.g., `bartowski:llama3.1-7b-gguf/model_id_xxx`)
4. Full path (e.g., `huggingface.co:bartowski/llama3.1-7b-gguf/model_id_xxx`)

It returns the shortest unique alias, appending a numeric suffix if necessary to ensure uniqueness.

### PR DESCRIPTION
**Pull request by vansangpfiev**
## Describe Your Changes
Use `sqlite` database to replace `model.list`
- Library: `SQLiteCpp`
- Keep the data structure the same as `model.list`
  - model_id: Unique identifier for the model
  - author_repo_id: Author or repository identifier
  - branch_name: Branch name in the repository
  - path_to_model_yaml: Path to the model's YAML file
  - model_alias: Shortened alias for the model
 - Use `cpp:result` for return value

## Fixes Issues

- Closes https://github.com/janhq/cortex.cpp/issues/1241

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/commands/model_import_cmd.cc [hunk 1]
```diff
@@ -16,15 +16,15 @@ ModelImportCmd::ModelImportCmd(std::string model_handle, std::string model_path)
...
 
   std::string model_yaml_path = (file_manager_utils::GetModelsContainerPath() /
                                  std::filesystem::path("imported") /
                                  std::filesystem::path(model_handle_ + ".yml"))
                                     .string();
-  modellist_utils::ModelEntry model_entry{
+  cortex::db::ModelEntry model_entry{
       model_handle_,   "local",       "imported",
-      model_yaml_path, model_handle_, modellist_utils::ModelStatus::READY};
+      model_yaml_path, model_handle_, cortex::db::ModelStatus::READY};
```
Comment (namchuai) [inline engine/commands/model_import_cmd.cc]: maybe we should remove `cortex::db::ModelStatus::READY`.

File: engine/database/database.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#include <string>
+#include "SQLiteCpp/SQLiteCpp.h"
+#include "utils/file_manager_utils.h"
+
+namespace cortex::db {
+const std::string kDefaultDbPath =
+    file_manager_utils::GetCortexDataPath().string() + "/cortex.db";
+class Database {
+ public:
+  Database(Database const&) = delete;
```
Comment (namchuai) [inline engine/database/database.h:12]: nice

File: engine/database/models.cc [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+Models::Models(SQLite::Database& db) : db_(db) {
+  db_.exec(
+      "CREATE TABLE IF NOT EXISTS models ("
+      "model_id TEXT PRIMARY KEY,"
+      "author_repo_id TEXT,"
+      "branch_name TEXT,"
+      "path_to_model_yaml TEXT,"
+      "model_alias TEXT UNIQUE,"
+      "status TEXT);");
```
Comment (namchuai) [inline engine/database/models.cc]: maybe we should remove status here
If we have the record inside db, we know that it's ready. other model status (loaded or something) will be stored in memory.

**Review by namchuai — APPROVED** [review]
✅ looks good to me. Great work!

File: engine/commands/model_import_cmd.cc (hunk 1 above)
Comment (vansangpfiev) [reply engine/commands/model_import_cmd.cc]: Removed it

**Review by nguyenhoangthuan99 — APPROVED** [review]

**Comment by dan-menlo** [comment]
@gabrielle-ong We will need to account for this in the QA checklist - i.e. ensuring that model list etc is persisted correctly in sqlite

**Pull request merged**